src/wazuh_modules/wm_exec.c

```diff
@@ -600,58 +599,50 @@ void* reader(void *args) {
 // Add process group to pool
 
 void wm_append_sid(pid_t sid) {
-    int i;
 
-    w_mutex_lock(&wm_children_mutex);
-
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (!wm_children[i]) {
-            wm_children[i] = sid;
-            break;
-        }
-    }
+    pid_t * p_sid = NULL;
 
-    w_mutex_unlock(&wm_children_mutex);
+    os_calloc(1, sizeof(pid_t), p_sid);
+    *p_sid = sid;
 
-    if (i == WM_POOL_SIZE)
-        merror("Child process pool is full. Couldn't register sid %d.", (int)sid);
+    if(!OSList_AddData(wm_children_list, (void *)p_sid)) {
+        merror("Child process sid %d could not be registered.", sid);
+        os_free(p_sid);
+    }
 }
 
 // Remove process group from pool
 
 void wm_remove_sid(pid_t sid) {
-    int i;
 
-    w_mutex_lock(&wm_children_mutex);
+    OSListNode * node_it = NULL;
+    pid_t * p_sid = NULL;
 
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        if (wm_children[i] == sid) {
-            wm_children[i] = 0;
-            break;
+    OSList_foreach(node_it, wm_children_list) {
+        p_sid = (pid_t *)node_it->data;
+        if(p_sid && *p_sid == sid) {
+            OSList_DeleteThisNode(wm_children_list, node_it);
+            os_free(p_sid);
+            return;
         }
     }
-
-    if (i == WM_POOL_SIZE)
-        merror("Child process %d not found.", (int)sid);
-
-    w_mutex_unlock(&wm_children_mutex);
+    merror("Child process %d not found.", sid);
 }
 
 // Terminate every child process group. Doesn't wait for them!
 
 void wm_kill_children() {
     // This function may be called from a signal handler
 
-    int i;
     int timeout;
     pid_t sid;
+    pid_t * p_sid = NULL;
+    OSListNode * node_it = NULL;
 
-    w_mutex_lock(&wm_children_mutex);
-
-    for (i = 0; i < WM_POOL_SIZE; i++) {
-        sid = wm_children[i];
-
-        if (sid) {
+    OSList_foreach(node_it, wm_children_list) {
```

Comment:
Although we know OSList has mutexes for I/O. Here we are forking child processes to kill the pids. 
The previous code did lock a mutex in the parent process. This keeps the list safe.
But using For Each, we are not locking the list. Is this acceptable?

Reply:
Right, will fix!